graphix/gflow.py

```diff
@@ -461,3 +392,382 @@ def get_layers(l_k):
     for i in l_k.keys():
         layers[l_k[i]].append(i)
     return d, layers
+
+
+class GF2Solver:
+    """Solver for GF(2) linear equations"""
+
+    def __init__(self, graph, input, output):
```

Comment:
The solver implementation looks great, I was just wondering since the `GF2Solver()` is meant ot solve non-square systems in $GF(2)^{\otimes(n)}$, should we always restrict the input parameter to be an instance of `nx.Graph()` [^1] ?

[^1]: I see you have not typecasted the input, but I guess you still are expecting a graph.

Reply:
Thank you! @pafloxy 

As you point out, the solver doesn't need to restrict its input to `nx.Graph()`. Furthermore, it doesn't need graph information to solve a system of equations. I'll remove arguments for `__init__` and separate the `get_adjacency_matrix` method out of the class.